internal/testbed/integration/metrics_validation.go

```diff
@@ -45,18 +43,23 @@ func assertExpectedMetricsAreInReceived(t *testing.T, expected, actual []pmetric
 				metrics := scopeMetrics.At(j).Metrics()
 				for k := 0; k < metrics.Len(); k++ {
 					actualMetric := metrics.At(k)
-					assert.Contains(t,
+					hasEntry := assert.Contains(t,
```

Comment:
nit: we could use `require.Contains` here so we don't have to do the check for `hasEntry` below